_studio/mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_legacy_defaults.cpp

```diff
@@ -1438,15 +1438,23 @@ class GetDefault
         , const mfxVideoParam& par
         , eMFXHWType hw)
     {
+        auto fcc = par.mfx.FrameInfo.FourCC;
+
         bool bValid =
```

Comment:
We can make this fussy function simpler, by
if (!Check<mfxU16, MFX_CODINGOPTION_ON, MFX_CODINGOPTION_OFF>(par.mfx.LowPower))
  return par.mfx.LowPower;
the rest will assume !bValid.
It is up to committer.

Reply:
I like this idea and will change the commit. Thanks!

Reply:
@mgonchar I'm afraid we don't support #ifdef preprocessor options to build the code for a specific MSDK API. We used to do this some time ago (a couple of years) when driver teams could use old msdk code (they didn't update it often). If we have to support some old MSDK API for some customer it's better to create branch for it.
Moreover some MSDK API version may require some specific driver version with corresponding libva (ddi) structures. Is it possible to do it in driver?
Anyway we've never tested msdk functionality with api versions other than the latest one.
@xhaihao I can't approve this PR because nobody can confirm that it won't break our validations. We need to check first if we use changing default behavior anywhere. Besides if we change default behavior for 444 then we need to do the same for 422 (vme by default), SCC (vdenc by default), 12b (vme by default), right? And this is only if we try to distinguish HW by profiles but we can continue the same exercise basing on feature scope.

Reply:
@saosipov The current SDK has already used the right mode for 422 (vme), SCC (vdenc), 12bit (vme) by default. Only for 444, the SDK uses the wrong mode (vme) by default. User has to specify low-power mode for 444 encoding, which  which confuses me.

Reply:
> I can't approve this PR because nobody can confirm that it won't break our validations.

I don't know how did you test this feature in your validations. Per my testing, -qsv-ff (or -lowpower:on) must be used for 444 encoding when using sample_encode without this path, otherwise user will get MFX errors.

```
$> sample_encode h265 -hw -ayuv -i input.ayuv.352x288.yuv -o output.h265 -w 352 -h 288

[ERROR], sts=MFX_ERR_ABORTED(-12), SynchronizeFirstTask, SyncOperation failed at /home/xhh/graphics/ex/gstreamer/MediaSDK/samples/sample_encode/src/pipeline_encode.cpp:168

[ERROR], sts=MFX_ERR_ABORTED(-12), GetFreeTask, m_TaskPool.SynchronizeFirstTask failed at /home/xhh/graphics/ex/gstreamer/MediaSDK/samples/sample_encode/src/pipeline_encode.cpp:2188

[ERROR], sts=MFX_ERR_ABORTED(-12), Run, m_pmfxENC->EncodeFrameAsync failed at /home/xhh/graphics/ex/gstreamer/MediaSDK/samples/sample_encode/src/pipeline_encode.cpp:2492

[ERROR], sts=MFX_ERR_ABORTED(-12), main, pPipeline->Run failed at /home/xhh/graphics/ex/gstreamer/MediaSDK/samples/sample_encode/src/sample_encode.cpp:1660
```

So I presumed the LowPower flag is set via -qsv-ff (or -lowpower:on) in your validation, and my change shouldn't have impact to your validations unless you have some negative cases in your validation to expect for these MFX errors.

Reply:
@xhaihao Could you update PR according to what you promised to do and we will take it?
We discussed this change internally and confirmed there's no risk with it.
As for your example with sample_encoder we'll have  a look.
I think you should have got UNSUPPORTED error instead of ABORT

Reply:
@saosipov I updated the PR, please review it when you get a chance.